docs/docs/tutorial/part-4/index.mdx

```diff
@@ -0,0 +1,280 @@
+- Use the **`useStaticQuery`** hook to pull data into a "building-block" component.
+- Use the **`gatsby-source-filesystem`** plugin to pull data into your site from your local file system.
+- Create an **MDX** file with front matter.
+
+## Meet Gatsby's GraphQL data layer
+
+One of the really powerful features of Gatsby is that you can pull data into your site from anywhere. For example, you might want to store your blog content in WordPress and your products in Shopify. With Gatsby, you can use the best tool for each type of data in your site.
+
+But how does this work under the hood? Gatsby has its own GraphQL **data layer** where it keeps all the data for your site. You can add special plugins to your site called **source** plugins. When you build your site, each source plugin will pull data from a particular place into the GraphQL data layer for your site.
+
+There are different source plugins for a variety of different data sources. You can find source plugins for your file system, several different content management systems (CMSs), private APIs, or databases.
+
+Once the data is in the data layer, you can write GraphQL queries inside of pages or components to pull out the data you want to use in your site.
+
+![Source plugins pull data out of a particular data source and into the GraphQL data layer for your site. You can write GraphQL queries to pull data out of the data layer and into your React components.](./data-layer.png)
+
+<Announcement>
+
+In this Tutorial, we'll teach you all the GraphQL you'll need to know to build your first Gatsby site. Interested in learning more? [How To GraphQL](https://www.howtographql.com/) is a free tutorial that teaches you the fundamentals.
+
+</Announcement>
+
+## Use GraphiQL to explore the data layer and write GraphQL queries
+
+How do you know what data is in your site's GraphQL data layer? When you run your site in develop mode, Gatsby automatically creates a special endpoint that lets you use **GraphiQL**. GraphiQL is an in-browser tool that you can use to explore your site's data and build GraphQL queries.
+
+You can view GraphiQL by going to `http://localhost:8000/___graphql` in your browser. (That's three underscores in the URL.)
+
+![A screenshot of the GraphiQL interface in a web browser. It has three main sections: the Explorer, the Query Editor, and the Result Window.](./graphiql.png)
+
+There are three main sections of the GraphiQL interface:
+
+* **Explorer:** This shows you all the different kinds of data you can request in a GraphQL query.
+    * You can toggle the dropdowns to expand the different fields and see what kinds of data is available in the data layer.
+* **Query Editor:** This is the middle section, which you can use to write out a query to test.
+    * You can add fields to your query by checking the boxes for different fields in the Explorer pane. Or, if you'd prefer, you can type the fields directly into the Query Editor.
```

Comment:
Maybe add that it has autocompletion with Ctrl + Space?

Reply:
Pro tip! I didn't even know you could do that 👏